modules/twinklexfd.js

```diff
@@ -1067,49 +1067,64 @@ Twinkle.xfd.callbacks = {
 				Twinkle.xfd.currentRationale = null;  // any errors from now on do not need to print the rationale, as it is safely saved on-wiki
 			});
 		},
-		userNotification: function(pageobj) {
+		courtesyNotification: function(pageobj) {
 			var initialContrib = pageobj.getCreator();
+			var inCategories = mw.config.get('wgCategories');
+			var pagesToNotify = [];
 			var params = pageobj.getCallbackParameters();
 
-			// Disallow warning yourself
-			if (initialContrib === mw.config.get('wgUserName')) {
+			// Notify initial contributor (if it's not the nominator)
+			if (initialContrib !== mw.config.get('wgUserName')) {
+				pagesToNotify.push('User talk:' + initialContrib);
+			} else {
 				pageobj.getStatusElement().warn('You (' + initialContrib + ') created this page; skipping user notification');
-				return;
 			}
 
-			var usertalkpage = new Morebits.wiki.page('User talk:' + initialContrib, 'Notifying initial contributor (' + initialContrib + ')');
-			var notifytext = '\n';
-			var modNotice = mw.config.get('wgPageContentModel') === 'Scribunto' ? '|module=yes' : '';
-			switch (params.xfdcat) {
-				case 'tfd':
-					notifytext += '{{subst:Tfd notice|1=' + mw.config.get('wgTitle') + modNotice + '}} ~~~~';
-					break;
-				case 'tfm':
-					notifytext += '{{subst:Tfm notice|1=' + mw.config.get('wgTitle') + '|2=' + params.target + modNotice + '}} ~~~~';
-					break;
-				default:
-					alert('twinklexfd in userNotification: unknown TFD action');
-					break;
+			// Notify developer(s) of script(s) that use(s) the nominated template
+			if (inCategories.includes('Templates used by Twinkle')) {
+				pagesToNotify.push('Wikipedia talk:Twinkle');
+			}
+			if (inCategories.includes('Templates used by AutoWikiBrowser')) {
+				pagesToNotify.push('Wikipedia talk:AutoWikiBrowser');
+			}
+			if (inCategories.includes('Templates used by RedWarn')) {
+				pagesToNotify.push('Wikipedia talk:RedWarn');
 			}
 
-			usertalkpage.setAppendText(notifytext);
-			usertalkpage.setEditSummary('Notification: [[' + params.discussionpage + '|listing]] of [[:' + pageobj.getPageName() + ']] at [[WP:TFD|templates for discussion]].' + Twinkle.getPref('summaryAd'));
-			usertalkpage.setCreateOption('recreate');
-			Twinkle.xfd.setWatchPref(usertalkpage, Twinkle.getPref('xfdWatchUser'));
-			usertalkpage.setFollowRedirect(true);
+			pagesToNotify.forEach(function(pagename, i) {
+				var pageToNotify = new Morebits.wiki.page(pagename, 'Notifying of template nomination');
+				var notifytext = '\n';
+				var modNotice = mw.config.get('wgPageContentModel') === 'Scribunto' ? '|module=yes' : '';
```

Comment:
Can you put the `notifytext` generation code outside the loop, as it needs to be evaluated just once?

Reply:
Done.